### PR TITLE
test: Backport rhel-7-4 testing fixes

### PR DESCRIPTION
**Pull request by dperpeet**
- [x] #7064

### CODE REVIEW

**Comment by dperpeet** [comment]
We can add the setup files for the 7.4 image later

**Comment by dperpeet** [comment]
Long-term it makes sense to fix this branch to conform with the testing/naughty setup on master.

**Comment by dperpeet** [comment]
After talking with @martinpitt there might be a better approach, work in progress

**Comment by martinpitt** [comment]
The tests that this PR touches passed, so this at least makes things incrementally better in the rhel-7.4 branch. I would be fine with merging it as-is, unless @stefwalter vetos?

**Comment by stefwalter** [comment]
> The tests that this PR touches passed, so this at least makes things incrementally better in the rhel-7.4 branch. I would be fine with merging it as-is, unless @stefwalter vetos?

Hmmm, but how do we dig ourselves out of the current hole of broken tests? I feel like I prematurely merged #6360 ... @dperpeet Are you creating a follow up to this?

**Comment by dperpeet** [comment]
> Hmmm, but how do we dig ourselves out of the current hole of broken tests? I feel like I prematurely merged #6360 ... @dperpeet Are you creating a follow up to this?

The tests are good on master, I'm working on backporting the correct testing fixes to this branch.

File: test/images/candlepin [hunk 1]
```diff
@@ -1 +1 @@
-candlepin-392368746586c79a7bb5aa6fa38de0b18121d51a.qcow2
\ No newline at end of file
+candlepin-29a74a0221a854a68e1b8e761f9e74c6a30654c0.qcow2
```
Comment (stefwalter) [inline test/images/candlepin]: Testing doesn't use the images on these branches, only from master. Did these changes slip in here by accident or is there a reason to change the images on the rhel-7-4 branch?

File: test/images/scripts/candlepin.bootstrap [hunk 2]
```diff
@@ -18,4 +18,4 @@
 # 02110-1301 USA.
 
 BASE=$(dirname $0)
-$BASE/virt-builder-fedora "$1" fedora-22 x86_64
+$BASE/virt-builder-fedora "$1" centos-7.3 x86_64
```
Comment (stefwalter) [inline test/images/scripts/candlepin.bootstrap]: Ditto above comment.

File: test/images/candlepin (hunk 1 above)
Comment (dperpeet) [inline test/images/candlepin]: changed

File: test/images/scripts/candlepin.bootstrap (hunk 2 above)
Comment (dperpeet) [inline test/images/scripts/candlepin.bootstrap]: changed

**Comment by dperpeet** [comment]
> Lets remove anything to do with images or recreating them. Images are consumed by master ... and downstream RHEL testing uses their own images. This generates unnecessary churn in the patches generated from this branch.

As discussed on IRC, I removed the image symlinks from this branch entirely. Image creation doesn't hurt, so I left that part in - removing the files would add long patches.

File: test/verify/files/openshift.kubeconfig [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
...U1F3SWdZRFZRUUREQnR2Y0dWdWMyaHBablF0YzJsbmJtVnlRREUwT1RBMk9ERTJNekV3CmdnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUURUYzNlQkRpQnZwUzFWWFo2VEhkaFAKYWJ4RENuSFdsb2VTZDMveERqaVBlRmpVN1RKeTJaUzNGT3J1cUVBTytKdGJWd0RhcU5VNkNEV1hXQ0FtK0dtdApBcE5YcG5hUlBVNDk2Vmx1b2RZZUV2a0FmWXdzZjM1a0s5TWxYbDUvY1AwYUE1UDNCbWRpT2kzdlBwcmxJeUtCClpIbzhSVU1YaXJyaHI0QjYzY2NKNHJMaDFQR28yR0tudFJyLzNnc0ZuRGlEQzFsTEZERDRIUU5GRnlTWFZRL3EKVHlxY1NZR3V2OGprRnQzem9nOENhckpHSG1PTzVYL25HY214OTJObUpTNHUwREJnRXNRaFhINXZadVBGQ3p6cQpVZktWc1M1RjJJL3AzdVRoK0RSUXYyMFZ1VTNCUUJhREplZkpUUUJ4eW9uYW5hRmg5S2YwV3gxb3dDdWtCM1dMCkFnTUJBQUdqSXpBaE1BNEdBMVVkRHdFQi93UUVBd0lDcERBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUEwR0NTcUcKU0liM0RRRUJDd1VBQTRJQkFRQnJ5elFaemFkc3FFL0JEY1VlVUE2d21QUHk0MGMyOXVDazlPRzdPL0M4WGFpNwpSdWxDMXlzcmZ4ZksxMExLNzNGcEM1aE93azJxTVBvQVBNTW9HSVNkSjVTem1WblZYMTNEVVI3NGN1dktsYnhDClZDVmE2OE1LU0pFZlZSRTY0SGE5TFdDVG1TaXJyemFKMUYvdGdwbkc2NnFSdDhMalA0TEY4STgwWGNXVEFEU3cKMHJRdGp0bFFEKy9aVVFLNEptWEQ3V1BjVnRUZWhON0x5WjcvR0VvVVRLR0lQMFZScldndm5iOHNIWEEwQUF0aApFS2ROVk9jUVdTVTlsM000MVFhM1R1eTRCSGhoTVZNUmZ1Y2MzWVZIMFdFVzBacU1JNFgxSEhjNGtjSEdWVXpDCllJRU1OcTdFZlpZeksyMmVsSUkxU2trR3o1WDRrUGtqZkdydFpHV1IKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
```
Comment (stefwalter) [inline test/verify/files/openshift.kubeconfig]: This file does not need to be updated in this pull request.

File: test/verify/check-subscriptions [hunk 4]
```diff
@@ -202,10 +225,10 @@ class TestSubscriptions(MachineCase):
         # enter server data
         b.wait_present("#subscription-register-url-custom")
         b.wait_visible("#subscription-register-url-custom")
-        b.set_val("#subscription-register-url-custom", "{0}:8443/candlepin".format(self.candlepin.address))
+        b.set_val("#subscription-register-url-custom", "{address}:8443/candlepin".format(address=self.candlepin.address))
```
Comment (stefwalter) [inline test/verify/check-subscriptions]: This change just looks like churn. Are we sure we want it the code generating a patch block for this?

File: test/verify/check-subscriptions [hunk 5]
```diff
@@ -114,7 +134,7 @@ class TestSubscriptions(MachineCase):
         # enter server and incorrect login data
         b.wait_present("#subscription-register-url-custom")
         b.wait_visible("#subscription-register-url-custom")
-        b.set_val("#subscription-register-url-custom", "{0}:8443/candlepin".format(self.candlepin.address))
+        b.set_val("#subscription-register-url-custom", "{address}:8443/candlepin".format(address=self.candlepin.address))
```
Comment (stefwalter) [inline test/verify/check-subscriptions]: This change just looks like churn. Are we sure we want it the code generating a patch block for this?

File: test/verify/files/openshift.kubeconfig (hunk 3 above)
Comment (dperpeet) [inline test/verify/files/openshift.kubeconfig]: The image changed on master, so this test here has to be updated. We'll need a follow-up to fix the issue on master I think.

File: test/verify/check-subscriptions (hunk 5 above)
Comment (dperpeet) [inline test/verify/check-subscriptions]: This way we can take the entire original commit - and it makes looking for changes easier in the future, so I think it's ok in this instance.

File: test/verify/check-subscriptions (hunk 4 above)
Comment (dperpeet) [inline test/verify/check-subscriptions]: This way we can take the entire original commit - and it makes looking for changes easier in the future, so I think it's ok in this instance.

**Comment by martinpitt** [comment]
#7064 landed, retrying.

**Comment by dperpeet** [comment]
> #7064 landed, retrying.

Thanks, I added the changes there to the openshift test fix commit in this pr.

**Comment by stefwalter** [comment]
Merged into rhel-7.4 branch.

**Comment by stefwalter** [comment]
Merged as b0632fe1357c73922c9c42b4fede77386932e234

**Pull request closed without merging**